oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -97,46 +102,46 @@ public StorageEsInstaller(ModuleManager moduleManager, int indexShardsNumber, in
         }
     }
 
-    private JsonObject createSetting(boolean record) {
-        JsonObject setting = new JsonObject();
-        setting.addProperty("index.number_of_shards", indexShardsNumber);
-        setting.addProperty("index.number_of_replicas", indexReplicasNumber);
-        setting.addProperty("index.refresh_interval", record ? TimeValue.timeValueSeconds(10).toString() : TimeValue.timeValueSeconds(indexRefreshInterval).toString());
-        setting.addProperty("analysis.analyzer.oap_analyzer.type", "stop");
+    protected Map<String, Object> createSetting(boolean record) {
+        Map<String, Object> setting = new HashMap<>();
+        setting.put("index.number_of_shards", indexShardsNumber);
+        setting.put("index.number_of_replicas", indexReplicasNumber);
+        setting.put("index.refresh_interval", record ? TimeValue.timeValueSeconds(10).toString() : TimeValue.timeValueSeconds(indexRefreshInterval).toString());
+        setting.put("analysis.analyzer.oap_analyzer.type", "stop");
```

Comment:
setting.put("index.max_result_window", 1000000);

As mentioned in the FAQ document, we set this option default.